packages/core/aqua-to-js/src/common.ts

```diff
@@ -35,7 +35,7 @@ export function genTypeName(
       const args =
         item.tag === "labeledProduct" ? Object.values(item.fields) : item.items;
 
-      if (args.length === 1) {
+      if ("0" in args) {
```

Comment:
these are actually different. Previously you checked the length is exactly equal to one. Now you are checking that first element exists. Does it make sense to still check the the length is exactly === 1. Can it ever be !== 1? If this is something aqua compiler enforces - I would probably check the case when it's not 1 and maybe throw error - this we will get a chance to discover some unexpected bug

the same question to think about is for the same check you are doing bellow

Reply:
Wil recheck these in other files